woocommerce/woocommerce-ios
# Blaze: Fix issue mapping campaign list

Closes: #11526


## Description

This PR fixes the mapping issue for campaign list due to the change in API response:
- Updated the `BlazeCampaign` model: 
  - Replaced the mapping of `target_urn` with `target_url` and replaced property `productID` with `productURL`
  - Replaced the mapping of `total_budget` with `budget_cents` and renamed the property `totalBudget` to `budgetCents`.
- Updated the conversion between storage and read-only models. We must update the storage model to use new properties in the next app version.
- Updated the Blaze campaign item to display budget by dividing cents by 100.
- Updated the check for product with existing campaigns by checking for product URL instead of product ID.

## Testing instructions

- Log in to a store that is eligible for Blaze.
- Create a new Blaze campaign if needed.
- Open the campaign list, make sure that the list is correct.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/efa1af1d-313c-4747-8cfb-ec76638d1091" width=320 />


---
